### PR TITLE
Retarget DOTNETHOME when installing x64 on ARM64

### PR DESCRIPTION
**Pull request by ericstj**
Draft that enables installing x64 product on ARM64.

Related PRs: 
https://github.com/dotnet/arcade/pull/7785
https://github.com/dotnet/runtime/pull/58669
https://github.com/dotnet/installer/pull/11843

### CODE REVIEW

**Comment by dougbu** [comment]
I don't think GitHub will notify us when this PR is ready for review. Could you please let us know @ericstj❔

**Comment by ericstj** [comment]
Sure, I'll let you know, mostly I just need to wait until the arcade change is reviewed and then replicate feedback here.  One thing that's unique to ASP.NET is this: https://github.com/dotnet/aspnetcore/pull/36100/files#diff-b146853a2811958d2332489ea0501ea790ed1df7061172676d9c65a143d499b9R70

Can you have a look and let me know what you think?

**Comment by dougbu** [comment]
> Can you have a look and let me know what you think?

@wtgodbe or @joeloff, I think you have more context to answer @ericstj's question about https://github.com/dotnet/aspnetcore/pull/36100/files#diff-841d5a29fed63d4cad2012857ad278859c9fbfdb19a78c6899af34c6155d34daR69-R72

@ericstj on the other hand, we avoid `TODO` comments as much as possible. If we need a fix there, please either do it in this PR or file an issue and mention the issue in those comments.

**Comment by ericstj** [comment]
> @ericstj Eric St. John FTE on the other hand, we avoid TODO comments as much as possible. If we need a fix there, please either do it in this PR or file an issue and mention the issue in those comments.

I'm not going to commit it. It's in this PR while in draft because I need feedback from ASP.NET about it.  I intend to get your feedback to address that comment and remove before committing.

**Comment by ericstj** [comment]
This should now be ready for review.  Note that I resolved the clashing registry keys between x64 and arm64 by making x64 install to a different path on ARM64 machines.

File: src/Installers/Windows/Common/dotnethome_x64.wxs
Comment (joeloff) [inline src/Installers/Windows/Common/dotnethome_x64.wxs:24]: Should this be a case-insensitive comparison?

Comment (ericstj) [reply src/Installers/Windows/Common/dotnethome_x64.wxs:24]: The values are documented explicitly as the strings defined previously in this file, case and all.   I did the mapping intentionally.

Also this goes away once we have the wix feature where will compare against numeric values for image file machine.

File: src/Installers/Windows/Common/dotnethome_x64.wxs
Comment (dougbu) [inline src/Installers/Windows/Common/dotnethome_x64.wxs:32]: Could you clarify "no other native architecture can install ARM64 today" a bit❔ Is there just one `Program Files` folder on ARM64 machines❔

File: src/Installers/Windows/Common/dotnethome_x64.wxs
Comment (dougbu) [inline src/Installers/Windows/Common/dotnethome_x64.wxs:33]: Shouldn't this be
```suggestion
    <SetProperty Action="Set_DOTNETHOME_NON_NATIVE_ARCHITECTURE" Id="DOTNETHOME" Value="[ProgramFiles64Folder]dotnet\" After="Set_NON_NATIVE_ARCHITECTURE">
```
❔

**Review by dougbu — COMMENTED** [review]
Couple of questions but I defer to @joeloff and @wtgodbe because they have more WiX-iness

File: src/Installers/Windows/Common/dotnethome_x64.wxs
Comment (ericstj) [reply src/Installers/Windows/Common/dotnethome_x64.wxs:33]: No, when x64 installer runs on a machine that isn't x64, we need to install to an x64 subfolder per https://github.com/dotnet/designs/blob/main/accepted/2021/x64-emulation-on-arm64/x64-emulation.md#install-location

File: src/Installers/Windows/Common/dotnethome_x64.wxs
Comment (ericstj) [reply src/Installers/Windows/Common/dotnethome_x64.wxs:32]: > Could you clarify "no other native architecture can install ARM64 today" a bit

So there are 3 architectures we support: x86, x64, and ARM64.  X86 does not need any redirection, since the OS does this for us.  Arm64 doesn't need any redirection since there is no other architecture that can install an ARM64 MSI (nor can we predict if one were created how the OS would treat it).  X64 is the only one that needs our own redirection.  See additional response.

> Is there just one Program Files folder on ARM64 machines❔

On ARM64 machines both x64 and arm64 installers get a single `Program Files` that they need to share.  There is still `Program Files (x86)` for x86 installers, but it is the responsibility of the installer itself to deal with side-by-side on ARM64.

**Comment by ericstj** [comment]
I believe I responded to all feedback, reviewers, can you please have a look at the responses and let me know if this looks good?

**Review by joeloff — APPROVED** [review]

**Comment by dougbu** [comment]
@ericstj do we need this fix in release/6.0❔ If yes, release/6.0-rc2 as well❔

**Comment by ericstj** [comment]
Yes, I will be bringing a long list of commits to port to 6.0-rc2

**Pull request merged**
